arroyo/backends/local/backend.py

```diff
@@ -314,16 +314,8 @@ def seek(self, offsets: Mapping[Partition, int]) -> None:
 
     def stage_offsets(self, offsets: Mapping[Partition, int]) -> None:
         with self.__lock:
-            if self.__closed:
-                raise RuntimeError("consumer is closed")
-
-            if offsets.keys() - self.__offsets.keys():
-                raise ConsumerError("cannot stage offsets for unassigned partitions")
-
-            self.__validate_offsets(
-                {partition: offset for (partition, offset) in offsets.items()}
-            )
-
+            # XXX: can we remove the locking? dictionary updates might be
+            # atomic
```

Comment:
Probably, the Kafka backend doesn't do this locking - messages need to be processed and committed in order. This one is just for testing anyway.